toolsets/src/ubuntu.cpp

```diff
@@ -167,3 +170,33 @@ void UbuntuToolset::CreateCommandsFor(Project &project)
         }
     }
 }
+
+GlobalOptions UbuntuToolset::ParseGeneralOptions(const std::string &options)
```

Comment:
The one design question I still don't know the answer to is whether the toolset should fix the compile / link options, or whether there'll be one set you have that this code then interpolates to how to tell it to your compiler. You've set it up very well for the second one, but I don't know which of the two it should be. Do you have an opinion & rationale?